### DISCUSSION TITLE
Request for soft inpainting

### DISCUSSION THREAD

**Discussion by tyomalu** [Acly/krita-ai-diffusion#451] [Ideas]
Auto1111 prerelease has [soft inpainting feature](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/14208). It's so much better than ordinary inpainting. Your plugin is fantastic! I wish this feature could be implemented.

**Comment by Acly** [comment]
Latest ComfyUI has a "Differential Diffusion" node which does something very similar (same idea).

While it's a substantial improvement over "naive" inpainting, it has little to no effect on how it's currently implemented in the plugin. That's for most use cases where soft alpha is used for smoother transitions at mask boundary. 

The exception would be for scenarios where you assign different noise strength to areas in a mask or some kind of gradient.

On the plus side, it doesn't seem to hurt performance.

**Comment by Acly** [comment]
Implemented in [version 1.16.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.16.0)
